interp/builtin.go

```diff
@@ -984,14 +984,16 @@ func (r *Runner) readLine(ctx context.Context, raw bool) ([]byte, error) {
 	}
 }
 
-func (r *Runner) changeDir(ctx context.Context, path string) uint8 {
+func (r *Runner) changeDir(ctx context.Context, cmd, path string) uint8 {
 	path = cmp.Or(path, ".")
 	path = r.absPath(path)
 	info, err := r.stat(ctx, path)
 	if err != nil || !info.IsDir() {
+		r.errf("%s: the directory '%s' does not exist\n", cmd, path)
```

Comment:
Even we don't need to be 100% compatible, I'd still try to have the same error format as it's easy enough.

```suggestion
		r.errf("%s: no such file or directory: %s\n", cmd, path)
```